geo/src/algorithm/euclidean_distance.rs

```diff
@@ -480,47 +397,248 @@ where
+impl_euclidean_distance_for_iter_geometry!(MultiPoint<T>, [Point<T>, Line<T>, LineString<T>, MultiLineString<T>, Polygon<T>, MultiPolygon<T>, GeometryCollection<T>]);
+impl_euclidean_distance_for_iter_geometry!(MultiLineString<T>, [Point<T>, Line<T>, LineString<T>, Polygon<T>, MultiPolygon<T>, GeometryCollection<T>]);
+impl_euclidean_distance_for_iter_geometry!(MultiPolygon<T>, [Point<T>, Line<T>, LineString<T>, Polygon<T>, GeometryCollection<T>]);
+impl_euclidean_distance_for_iter_geometry!(GeometryCollection<T>, [Point<T>, Line<T>, LineString<T>, Polygon<T>]);
+impl_euclidean_distance_for_iter_geometry!(MultiPoint<T>);
+impl_euclidean_distance_for_iter_geometry!(MultiLineString<T>);
+impl_euclidean_distance_for_iter_geometry!(MultiPolygon<T>);
+impl_euclidean_distance_for_iter_geometry!(GeometryCollection<T>);
+
+/// Implements Euclidean distance for Triangles and Rects by converting them to polygons.
+macro_rules! impl_euclidean_distance_for_polygonlike_geometry {
+  ($for:ty,  [$($target:ty),*]) => {
+      $(
+          impl<T> EuclideanDistance<T, $target> for $for
+          where
+              T: GeoFloat + Signed + RTreeNum + FloatConst,
+          {
+              fn euclidean_distance(&self, other: &$target) -> T {
+                  other.euclidean_distance(&self.to_polygon())
+              }
+          }
+
+          impl<T> EuclideanDistance<T, $for> for $target
+          where
+              T: GeoFloat + Signed + RTreeNum + FloatConst,
+          {
+              fn euclidean_distance(&self, other: &$for) -> T {
+                  self.euclidean_distance(&other.to_polygon())
+              }
+          }
+      )*
+  };
+  ($for:ty) => {
+    impl<T> EuclideanDistance<T, $for> for $for
+    where
+        T: GeoFloat + FloatConst + RTreeNum + FloatConst,
+    {
+        fn euclidean_distance(&self, other: &$for) -> T {
+            other.euclidean_distance(&self.to_polygon())
+        }
     }
+  };
 }
 
+impl_euclidean_distance_for_polygonlike_geometry!(Triangle<T>, [Point<T>, MultiPoint<T>, Line<T>, LineString<T>, MultiLineString<T>, Polygon<T>, MultiPolygon<T>, GeometryCollection<T>, Rect<T>]);
+impl_euclidean_distance_for_polygonlike_geometry!(Rect<T>, [Point<T>, MultiPoint<T>, Line<T>, LineString<T>, MultiLineString<T>, Polygon<T>, MultiPolygon<T>, GeometryCollection<T>]);
+impl_euclidean_distance_for_polygonlike_geometry!(Triangle<T>);
+impl_euclidean_distance_for_polygonlike_geometry!(Rect<T>);
+
 // ┌──────────────────────────────┐
-// │ Implementations for Triangle │
+// │ Implementation for Geometry  │
 // └──────────────────────────────┘
 
-impl<T> EuclideanDistance<T, Point<T>> for Triangle<T>
+impl<T> EuclideanDistance<T> for Geometry<T>
 where
-    T: GeoFloat,
+    T: GeoFloat + FloatConst,
 {
-    fn euclidean_distance(&self, point: &Point<T>) -> T {
-        if self.intersects(point) {
-            return T::zero();
+    fn euclidean_distance(&self, other: &Geometry<T>) -> T {
```

Comment:
My condolences to your wrists. 😉 

I'd expect this to be implementable as
```
impl<T> EuclideanDistance<T> for Geometry<T>
 where T: GeoFloat + FloatConst 
{
    crate::geometry_delegate_impl! {
        fn euclidean_distance(&self, other: &Geometry<T>) -> T;
    }
}
```

Please revisit this method after addressing the missing implementations mentioned above.

Reply:
thanks! done